upgrade/upgrade-scripts/common.sh

```diff
@@ -483,5 +483,6 @@ function fix_and_migrate_services() {
 		snmpd.service
 		systemd-timesyncd.service
 		td-agent.service
+		telegraf.service
```

Comment:
hm.. I'm not against this, but I'm curious..

since the `performance-diagnostics` package depends on `telegraf`.. should `performance-diagnostics` be responsible for masking the service? would that work? e.g. if we masked it in the postinst script for `performance-diagnostics`..

I think I'm OK with this, but I prefer not to rely on the upgrade scripts to compensate for "bugs" in other packages.. IMO, I'd argue, if it's the `performance-diagnostics` package that results in the `telegraf` service getting pulled in and enabled incorrect, it should be up to that package to fix that broken state..

Further, does this actually work? What testing have we done? From what I can tell, this `fix_and_migrate_services` function is called before we upgrade packages.. so if it's an upgrade of the `telegraf` package that results in this service getting enabled, I feel like this change might not fix the situation.. since it'll mask it prior to the package upgrade, meaning when the package is later upgraded, it'll re-enable the service..

Thoughts?

Reply:
Hmm, but that same argument can be made about almost every service on this list. For example, td-agent.service. The delphix-mgmt service uses td-agent, and needs the generic td-agent.service to be masked for the exact same reason that telegraf.service needs to be masked (the bundled service is not used).

Reply:
As mentioned in the meeting.. it looks like the `telegraf` package will `systemctl enable` the `telegraf` service..

So, I think masking the service via the `performance-diagnostic` package's postinst script would work..

But, I think this problem is similar to other services (e.g. `postgres`, `td-agent`, etc.).. so, to be consistent with how we've solved it for those services, I think what you have here is best.

Thus, I'll resolve this comment, and approve the review as is.

Reply:
Yeah, we just spoke about this in the call as well. It would have been ideal to do this via respective postinst scripts but to remain consistent, we are going to mask that service here.

Reply:
I don't think that those services can be "unmasked" by the enabling of the service by any other package's actions (unless they explicitly unmask, which is very unlikely). The reason why doing the masking of services here seems more correct than in any postinst script is that if you do it in a postinst script, then the service may start briefly before then being masked, whereas doing it before the upgrade ensures that the package upgrade step doesn't enable the service at all.

Reply:
> then the service may start briefly before then being masked,

yea, good point.. maybe that's why we've chosen to do it this way.. I couldn't recall why, but that might be it and makes sense.

> but that same argument can be made about almost every service on this list.

yea, I agree after thinking about it a little more.